addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -243,14 +243,22 @@ export class PaymentScreen extends Component {
 
         // 1. Save order to server.
         this.env.services.ui.block()
-        const syncOrderResult = await this.pos.push_single_order(this.currentOrder);
-        this.env.services.ui.unblock()
-
-        if (syncOrderResult instanceof ConnectionLostError) {
-            this.pos.showScreen(this.nextScreen);
-            return;
-        } else if (!syncOrderResult) {
-            return;
+        let syncOrderResult;
+        try {
+            syncOrderResult = await this.pos.push_single_order(this.currentOrder);
+            if (!syncOrderResult) {
+                return;
+            }
+        } catch(error) {
+            if (error instanceof ConnectionLostError) {
+                this.pos.showScreen(this.nextScreen);
+                Promise.reject(error);
+                return error;
+            } else {
+                throw error;
+            }
+        } finally {
+            this.env.services.ui.unblock()
```

Comment:
there is a try-catch below for the download of invoice. I think you can reuse is because invoicing error because of connection lost should also show the next screen.